File: 2269. Find the K-Beauty of a Number/solution.py
class Solution(object):
  def divisorSubstrings(self, num, k):
    """
    :type num: int
    :type k: int
    :rtype: int
    """
    strNum = str(num)
    result = 0
    
    for index in range(0, len(strNum) - k + 1):
      value = int(strNum[index:index + k])
      if value != 0 and num % value == 0:
        result += 1
        
    return result

File: 2269. Find the K-Beauty of a Number/solution.cpp
class Solution {
public:
  int divisorSubstrings(int num, int k) {
    string strNum = to_string(num);
    int result = 0;

    for (int index = 0; index < strNum.size() - k + 1; index++){
      int value = stoi(strNum.substr(index, k));
      if (value != 0 && num % value == 0){
        result += 1;
      }
    }

    return result;
  }
};